source/module_hamilt_lcao/module_gint/gtask_force.cpp

```diff
@@ -0,0 +1,259 @@
+                              const UnitCell& ucell,
+                              const int i,
+                              const int j,
+                              const int psiSizeMax,
+                              const int max_size,
+                              const int nczp,
+                              const double vfactor,
+                              double* rcut,
+                              const double* vlocal_global_value,
+                              int* iat_per_nbz,
+                              const int lgd,
+                              double* dm_matrix_g,
+                              int& max_m,
+                              int& max_n,
+                              int& atom_pair_num,
+                              SGridParameter& para)
+{
+    const int grid_index_ij = i * gridt.nby * gridt.nbzp + j * gridt.nbzp;
+    const int nwmax = ucell.nwmax;
+    bool* gpu_mat_cal_flag = new bool[max_size * gridt.nbzp];
+
+    for (int i = 0; i < max_size * gridt.nbzp; i++)
+    {
+        gpu_mat_cal_flag[i] = false;
+    }
+    // psir generate
+    for (int z_index = 0; z_index < gridt.nbzp; z_index++)
+    {
+        int num_get_psi = 0;
+        int grid_index = grid_index_ij + z_index;
+        int num_psi_pos = psiSizeMax * z_index;
+        int calc_flag_index = max_size * z_index;
+        int bcell_start_index = gridt.bcell_start[grid_index];
+        int na_grid = gridt.how_many_atoms[grid_index];
+
+        for (int id = 0; id < na_grid; id++)
+        {
+            int ib = 0;
+            int mcell_index = bcell_start_index + id;
+            int imcell = gridt.which_bigcell[mcell_index];
+            int iat = gridt.which_atom[mcell_index];
+            int it_temp = ucell.iat2it[iat];
+            int start_ind_grid = gridt.start_ind[grid_index];
+
+            for (int bx_index = 0; bx_index < gridt.bx; bx_index++)
+            {
+                for (int by_index = 0; by_index < gridt.by; by_index++)
+                {
+                    for (int bz_index = 0; bz_index < gridt.bz; bz_index++)
+                    {
+                        double dr_temp[3];
+                        dr_temp[0] = gridt.meshcell_pos[ib][0]
+                                     + gridt.meshball_positions[imcell][0]
+                                     - gridt.tau_in_bigcell[iat][0];
+                        dr_temp[1] = gridt.meshcell_pos[ib][1]
+                                     + gridt.meshball_positions[imcell][1]
+                                     - gridt.tau_in_bigcell[iat][1];
+                        dr_temp[2] = gridt.meshcell_pos[ib][2]
+                                     + gridt.meshball_positions[imcell][2]
+                                     - gridt.tau_in_bigcell[iat][2];
+                        /* compute distance in and allocate the paramter in
+                         * z_index */
+                        double distance = sqrt(dr_temp[0] * dr_temp[0]
```

Comment:
is this a const?

Reply:
The distance for calculating the spherical harmonic function is the square of the distance for the interpolation function, and they share a common variable, so the const qualifier cannot be used